### DISCUSSION TITLE
simple crossfade with liquidsoap v2.0.3

### DISCUSSION THREAD

**Discussion by masternet0** [savonet/liquidsoap#2344] [General]
I'm new with liquidsoap and installed latest version on ubuntu 20.04 
I use it with icecast configuration 
Need to use simple crossfade like this :

def get_next_song() =
  uri = list.hd(process.read.lines("/root/aa.sh"))
  request.create(uri)
end

radio=request.dynamic(id="plugin-input",get_next_song)
radio=crossfade(start_next=1., fade_in=5., fade_out=5., radio)

What the function can i use because i tested lots and have errors and i didnt see any example with latest version 2.0.3
because I see each version for liquidsoap use different syntax not same its so more confused 

I try to read this docs https://www.liquidsoap.info/doc-2.0.3/crossfade.html and not works with simple crossfade

Any help please ?
Regards

**Comment by rrolla** [comment]
@masternet0 I think this can help you https://github.com/savonet/liquidsoap/discussions/2100

**Reply by masternet0** [reply]
Thank you @rrolla 
but how can apply this in my output ?

```
def transition(a,b)
  if b.metadata["source"] != "music" then
    cross.smart(fade_in=0.5, fade_out=0.1, a, b)
  else
    # Else, apply the standard fade
    # cross.smart(fade_in=7., fade_out=3., a, b)
    cross.simple(fade_in=7., fade_out=3., a.source, b.source)
  end
end

def get_next_song() =
uri = list.hd(process.read.lines("/root/aa.sh"))
request.create(uri)
end

radio=request.dynamic(id="plugin-input",get_next_song)

output.icecast( %mp3.cbr(stereo=true,stereo_mode="default",samplerate=44100,bitrate=320,id3v2=true),
....
..... ..etc

```
in my radio how add this crossfade function to it
radio=request.dynamic(id="plugin-input",get_next_song)

**Reply by rrolla** [reply]
hey @masternet0 

You can add like this:

```rb
# Now a custom crossfade function:
def customTransition(a, b)
  # If new source is not music, fade just a bit
  if b.metadata["source"] != "music" then
    cross.smart(fade_in=.8, fade_out=.5, a, b)
  else
    # Else, apply the standard fade
    cross.simple(fade_in=7., fade_out=3., a.source, b.source)
  end
end

radio = cross(duration=5., customTransition, radio)

```
